meson.build

```diff
@@ -0,0 +1,822 @@
+    coverage_cflags   = ['-O0', '-g', '--coverage', '-U_FORTIFY_SOURCE']
+    common_ldflags   += ['--coverage']
+    common_cflags    += coverage_cflags
+    libmcount_cflags += coverage_cflags
+    test_cflags      += coverage_cflags
+endif
+
+if get_option('san') != ''
+    if get_option('san') == 'all'
+        sanitize_cflags = ['-O0', '-g', '-fsanitize=address,leak,undefined']
+        common_ldflags += ['-fsanitize=address,leak,undefined']
+    else
+        sanitize_cflags = ['-O0', '-g', '-fsanitize=@0@'.format(get_option('sanitize'))]
+        common_ldflags += ['-fsanitize=@0@'.format(get_option('sanitize'))]
+    endif
+
+    uftrace_cflags    += sanitize_cflags
+    demangler_cflags  += sanitize_cflags
+    symbols_cflags    += sanitize_cflags
+    dbginfo_cflags    += sanitize_cflags
+    traceevent_cflags += sanitize_cflags
+    test_cflags       += sanitize_cflags
+endif
+
+################################
+# -Dwith-capstone build option #
+################################
+
+libcapstone_is_enabled = false
+libcapstone_dep = dependency('capstone', required: get_option('with-capstone'))
+if libcapstone_dep.found()
+    libcapstone_is_enabled = true
+    common_cflags += '-DHAVE_LIBCAPSTONE'
+endif
+
+##############################
+# -Dwith-libdw  build option #
+# -Dwith-libelf build option #
+##############################
+
+libdw_is_enabled = false
+libdw_dep = []
+
+libelf_is_enabled = false
+libelf_dep = dependency('libelf', required: get_option('with-libelf'))
+if libelf_dep.found()
+    libelf_is_enabled = true
+    common_cflags += '-DHAVE_LIBELF'
+
+    libdw_dep = dependency('libdw', required: get_option('with-libdw'))
+    if libdw_dep.found()
+        libdw_is_enabled = true
+        common_cflags += '-DHAVE_LIBDW'
+    endif
+endif
+
+#################################
+# -Dwith-libluajit build option #
+#################################
+
+libluajit_is_enabled = false
+libluajit_dep = dependency('luajit', required: get_option('with-libluajit'))
+if libluajit_dep.found()
+    libluajit_is_enabled =true
+    common_cflags += '-DHAVE_LIBLUAJIT'
+endif
+
+###################################
+# -Dwith-libncursesw build option #
+###################################
+
+libncursesw_is_enabled = false
+libncursesw_dep = dependency('ncursesw', required: get_option('with-libncursesw'))
+if libncursesw_dep.found()
+    libncursesw_is_enabled = true
+    common_cflags += '-DHAVE_LIBNCURSES'
+endif
+
+#################################
+# -Dwith-libpython build option #
+#################################
+
+libpython_is_enabled = false
+libpython_dep = []
+python_cflags = []
+libpython2_dep = dependency('python2', required: get_option('with-libpython'))
+libpython3_dep = dependency('python3', required: get_option('with-libpython'))
+if libpython3_dep.found()
+    libpython_dep = libpython3_dep
+    libpython_is_enabled = true
+    common_cflags += '-DHAVE_LIBPYTHON3'
+    common_cflags += '-DLIBPYTHON_VERSION=3.8'
```

Comment:
Hmm.. I forgot to change this part by detecting the python3 version yet. I don't know how to do that in meson but will think about it more.

Reply:
Related part in the original build is at https://github.com/namhyung/uftrace/blob/master/check-deps/Makefile.check#L26.

Reply:
I it like this
```
find_python_version = 'import sysconfig; print(sysconfig.get_config_vars()["LDVERSION"],end="")'
...
    python_version = run_command(['python3', '-c', find_python_version])
    common_cflags += '-DLIBPYTHON_VERSION=@0@'.format(python_version.stdout())
```